p2p/p2p_test.go

```diff
@@ -214,3 +214,97 @@ func TestSend(t *testing.T) {
 		t.Fatalf("Did not receive message from %s", sa.hostAddr)
 	}
 }
+
+func TestGossiping(t *testing.T) {
+
+	//Start node A
+	nodeConfigA := &Config{
+		BootstrapNodes: nil,
+		Port:           7000,
+		NoBootstrap:    true,
+		NoMdns:         true,
+	}
+
+	nodeA, err := NewService(nodeConfigA, nil)
+	if err != nil {
+		t.Fatalf("Could not start p2p service: %s", err)
+	}
+
+	defer nodeA.Stop()
+
+	nodeA_Addr := nodeA.FullAddrs()[0]
+
+	//Start node B
+	nodeConfigB := &Config{
+		BootstrapNodes: []string{
+			nodeA_Addr.String(),
+		},
+		Port:   7001,
+		NoMdns: true,
+	}
+
+	msgChanB := make(chan Message)
+	nodeB, err := NewService(nodeConfigB, msgChanB)
+	if err != nil {
+		t.Fatalf("Could not start p2p service: %s", err)
+	}
+
+	defer nodeB.Stop()
+
+	nodeB_Addr := nodeB.FullAddrs()[0]
+
+	//Connect node A & node B
+	err = nodeB.bootstrapConnect()
+	if err != nil {
+		t.Errorf("Start error :%s", err)
+	}
+
+	//Start node C
+	nodeConfigC := &Config{
+		BootstrapNodes: []string{
+			nodeB_Addr.String(),
+		},
+		Port:   7002,
+		NoMdns: true,
+	}
+
+	msgChanC := make(chan Message)
+	nodeC, err := NewService(nodeConfigC, msgChanC)
+	if err != nil {
+		t.Fatalf("Could not start p2p service: %s", err)
+	}
+
+	defer nodeC.Stop()
+
+	//Connect node B & node C
+	err = nodeC.bootstrapConnect()
+	if err != nil {
+		t.Errorf("Start error :%s", err)
+	}
+
+	// Create mock BlockRequestMessage to broadcast
+	endBlock, err := common.HexToHash("0xfd19d9ebac759c993fd2e05a1cff9e757d8741c2704c8682c15b5503496b6aa1")
```

Comment:
Not a fan of magic values. Please document what input you used for this or just hash a concrete instance